### ISSUE TITLE
Apple Reviewers don't see in-app products but everything works in TestFlight

### ISSUE THREAD

**Issue by kbrmimbyl** [flutter/flutter#96223]
## Steps to Reproduce

1. Implement in-app purchases using `in_app_purchase: 0.5.0` and `flutter 2.0.0`
2. Submit to Apple for review
3. Apple rejects the app because they can't see the in-app-products but everything works perfectly in TestFlight

**Expected results:** 
Both I and Apple reviewers should see the in-app-products.

**Actual results:** 
I see the in-app products in TestFlight and can also make a purchase. Everything works perfectly. But Apple reviewers don't see the in-app-products, so they reject the app.

<details>
<summary>Code sample</summary>

```dart
  _initInAppPurchases() async {
    InAppPurchaseConnection.enablePendingPurchases();
    _iap = InAppPurchaseConnection.instance;
    iapAvailable = await _iap.isAvailable();
    if (iapAvailable) {
      // Get products
      Set<String> ids = Set.from([
        AppConstants().silverProductId,
        AppConstants().goldProductId,
      ]);
      ProductDetailsResponse response1 = await _iap.queryProductDetails(ids);
      iapProducts = response1.productDetails;

      // Listen to new purchases
      _iapSubscription = _iap.purchaseUpdatedStream.listen(
        (data) {
          print('NEW PURCHASE');
          setBuyingInProgress(false);
          iapPurchases.addAll(data);
          _verifyAndGiveGoodies(data[0]);
        },
      );
    }
  }
```

</details>

**Comment by darshankawar** [comment]
@kbrmimbyl 
I am not sure if this is specific to Flutter, but what rejection message do they give ?

Also check [this](https://developer.apple.com/forums/thread/120852) link and see if it helps.

**Comment by kbrmimbyl** [comment]
The issue is fixed. Sorry to bother everyone but I hope my learning helps. I raised this issue on the flutter repo to check if there's something in my code that's preventing Apple reviewers from viewing the IAP products. Turns out the answer is NO!!!

Apple's IAP review process seems to be far from working. They kept rejecting my binary saying they can't view the IAP, and finally approved it when I sent them a video of the thing working on TestFlight. Looks like they know that their review process isn't working and want to rely on video evidence from the developer :-)
